serve/graph/model/transaction.go

```diff
@@ -2,43 +2,228 @@ package model
 
 import (
 	"encoding/base64"
+	"encoding/json"
 	"fmt"
 
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	"github.com/gnolang/gno/tm2/pkg/amino"
 	"github.com/gnolang/gno/tm2/pkg/bft/types"
+	"github.com/gnolang/gno/tm2/pkg/sdk/bank"
+	"github.com/gnolang/gno/tm2/pkg/std"
 )
 
 type Transaction struct {
-	t *types.TxResult
+	memo     string
+	txResult *types.TxResult
+	messages []*TransactionMessage
 }
 
-func NewTransaction(t *types.TxResult) *Transaction {
-	return &Transaction{t: t}
+func NewTransaction(txResult *types.TxResult) *Transaction {
+	var stdTx std.Tx
+
+	if err := amino.Unmarshal(txResult.Tx, &stdTx); err != nil {
+		return nil
+	}
+
+	messages := make([]*TransactionMessage, 0)
+	for _, message := range stdTx.GetMsgs() {
+		messages = append(messages, NewTransactionMessage(message))
+	}
```

Comment:
Could we make this operation lazy? we can only apply the unmarshalling when needed, that is, when calling Messages() method. We can use `sync.Once` for that. Queries not reading messages will be much faster.

Reply:
I changed it to only decode when accessing stdTx and messages.